walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,70 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+       
+        // Initialization
+        ArrayList<TimeRange> TimeSegList = new ArrayList<TimeRange>();
+        ArrayList<TimeRange> Answer = new ArrayList<TimeRange>();
+        
+        // Find all invalid time ranges
+        for (Event nowEvent: events) {
+            
+            Set<String> eventAttendees = nowEvent.getAttendees();
+            Collection<String> requestAttendees = request.getAttendees();
+            boolean flag = false;
+
+            for (String nowAtt: requestAttendees) {
+                if (eventAttendees.contains(nowAtt)) {
+                    flag = true;
+                    break;
+                }
+            }
+
+            if (flag == true) {
+                TimeSegList.add(nowEvent.getWhen());
+            }
+
+            /* Here the use of retainAll seems to be wrong... but why?
+            eventAttendees.retainAll(request.getAttendees());
+            if (eventAttendees.size() > 0) {
+                TimeSegList.add(nowEvent.getWhen());
```

Comment:
I think that's because eventAttendees is an unmodifiable set.

Refer to public Set<String> getAttendees() in Event.java:

https://github.com/FromATP/my-portfolio/blob/portfolio_base/walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/Event.java#L76

Maybe you can use the Collections.disjoint method:

https://docs.oracle.com/javase/6/docs/api/java/util/Collections.html#disjoint%28java.util.Collection,%20java.util.Collection%29